src/django_components/component.py

```diff
@@ -303,106 +289,78 @@ def _fill_slots(
-        # To help with easy-to-overlook typos, we fuzzy match unresolvable fills to
-        # those slots for which no matching fill was encountered. In the event of
-        # a close match, we include the name of the matched unfilled slot as a
-        # hint in the error message.
-        #
-        # Note: Finding a good `cutoff` value may require further trial-and-error.
-        # Higher values make matching stricter. This is probably preferable, as it
-        # reduces false positives.
-        for fill_name in unmatched_fills:
-            fuzzy_slot_name_matches = difflib.get_close_matches(fill_name, unfilled_slots, n=1, cutoff=0.7)
-            msg = (
-                f"Component '{self.registered_name}' passed fill "
-                f"that refers to undefined slot: '{fill_name}'."
-                f"\nUnfilled slot names are: {sorted(unfilled_slots)}."
-            )
-            if fuzzy_slot_name_matches:
-                msg += f"\nDid you mean '{fuzzy_slot_name_matches[0]}'?"
-            raise TemplateSyntaxError(msg)
-
-        # Return updated FILLED_SLOTS_CONTEXT map
-        filled_slots_map: Dict[Tuple[SlotName, Template], FillContent] = {
-            (slot_name, template): content_data
-            for slot_name, content_data in slot_name2fill_content.items()
-            if content_data  # Slots whose content is None (i.e. unfilled) are dropped.
-        }
-        try:
-            prev_context: FilledSlotsContext = context[FILLED_SLOTS_CONTENT_CONTEXT_KEY]
-            return prev_context.new_child(filled_slots_map)
-        except KeyError:
-            return ChainMap(filled_slots_map)
+    def __repr__(self) -> str:
+        return "<ComponentNode: {}. Contents: {!r}>".format(
+            self.name_fexp,
+            getattr(self, "nodelist", None),  # 'nodelist' attribute only assigned later.
+        )
+
+    def render(self, context: Context) -> str:
+        resolved_component_name = self.name_fexp.resolve(context)
+        component_cls: Type[Component] = component_registry.get(resolved_component_name)
+
+        # Resolve FilterExpressions and Variables that were passed as args to the
+        # component, then call component's context method
+        # to get values to insert into the context
+        resolved_context_args = [safe_resolve(arg, context) for arg in self.context_args]
+        resolved_context_kwargs = {key: safe_resolve(kwarg, context) for key, kwarg in self.context_kwargs.items()}
+
+        if isinstance(self.fill_nodes, ImplicitFillNode):
+            fill_content = self.fill_nodes.nodelist
+        else:
+            fill_content = []
+            for fill_node in self.fill_nodes:
+                # Note that outer component context is used to resolve variables in
+                # fill tag.
+                resolved_name = fill_node.name_fexp.resolve(context)
+                resolved_fill_alias = fill_node.resolve_alias(context, resolved_component_name)
```

Comment:
The alias resolve logic from here https://github.com/EmilStenstrom/django-components/pull/416/files#diff-7345171ed38e1dd2e4b8199b65a2fd082323bdb331ff0ef0dc0dd3bd43f111aeL378 was moved to `NamedFillNode.resolve_alias`